implementation/src/main/java/io/smallrye/mutiny/operators/multi/MultiDemandPacer.java

```diff
@@ -0,0 +1,143 @@
+
+    public MultiDemandPacer(Multi<? extends T> upstream, ScheduledExecutorService executor, DemandPacer pacer) {
+        super(upstream);
+        this.executor = executor;
+        this.pacer = pacer;
+    }
+
+    @Override
+    public void subscribe(MultiSubscriber<? super T> subscriber) {
+        upstream.subscribe(new MultiSubscriptionPacerProcessor<>(subscriber, executor, pacer));
+    }
+
+    private static class MultiSubscriptionPacerProcessor<T> extends MultiOperatorProcessor<T, T> {
+
+        private final ScheduledExecutorService executor;
+        private final DemandPacer pacer;
+
+        private final AtomicLong itemsCounter = new AtomicLong();
+        private ScheduledFuture<?> scheduledFuture;
+        private DemandPacer.Request currentRequest;
+
+        MultiSubscriptionPacerProcessor(MultiSubscriber<? super T> downstream, ScheduledExecutorService executor,
+                DemandPacer pacer) {
+            super(downstream);
+            this.executor = executor;
+            this.pacer = pacer;
+        }
+
+        @Override
+        public Context context() {
+            if (downstream instanceof ContextSupport) {
+                return ((ContextSupport) downstream).context();
+            } else {
+                return Context.empty();
+            }
+        }
+
+        private void demandAndSchedule(ScheduledExecutorService executor) {
+            long demand = currentRequest.demand();
+            long delay = currentRequest.delay().toNanos();
+            scheduledFuture = executor.schedule(this::tick, delay, TimeUnit.NANOSECONDS);
+            upstream.request(demand);
+        }
+
+        private void tick() {
+            if (upstream == Subscriptions.CANCELLED) {
+                return;
+            }
+            long numberOfItemsEmitted = itemsCounter.getAndSet(0L);
+            try {
+                currentRequest = pacer.apply(currentRequest, numberOfItemsEmitted);
+                if (currentRequest == null) {
+                    cancel();
+                    downstream.onFailure(new NullPointerException("The pacer provided a null request"));
+                    return;
+                }
+            } catch (Throwable failure) {
+                cancel();
+                downstream.onFailure(failure);
+                return;
+            }
+            demandAndSchedule(executor);
+        }
+
+        @Override
+        public void onSubscribe(Subscription subscription) {
+            super.onSubscribe(subscription);
+            try {
+                currentRequest = pacer.initial();
+            } catch (Throwable failure) {
+                cancel();
+                downstream.onFailure(failure);
+                return;
+            }
+            if (currentRequest == null) {
+                cancel();
+                downstream.onFailure(new NullPointerException("The pacer provided a null initial request"));
+            } else {
+                demandAndSchedule(executor);
```

Comment:
All previous `onFailure()` calls would be safe if canceled. We can likely save an executor new task by adding another extra check.